jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -174,6 +177,15 @@ public final class System {
 		platformEncoding = getSysPropBeforePropertiesInitialized(sysPropID_PlatformEncoding);
 		String definedOSEncoding = getSysPropBeforePropertiesInitialized(sysPropID_OSEncoding);
 		/*[IF JAVA_SPEC_VERSION >= 18]*/
+
+		/*[IF PLATFORM-mz64]*/
+		/* As part of better handling of JEP400 constraints on z/OS, the com.ibm.autocvt property
+		 * determines whether we convert input file I/O based on file tagging. If not explicitly specified,
+		 * the property defaults to true, unless file.encoding is set to COMPAT.
+		 */
+		zOSAutoConvert = getSysPropBeforePropertiesInitialized(sysPropID_zOSAutoConvert);
+		/*[ENDIF] PLATFORM-mz64 */
```

Comment:
Suggest this should not be nested within the Java >= 18 but should be for Java >= 21; and should include `PLATFORM-mz31` for consistency with the native code:
```java
		/*[IF (JAVA_SPEC_VERSION >= 21) & (PLATFORM-mz31 | PLATFORM-mz64)]*/
		/* As part of better handling of JEP400 constraints on z/OS, the com.ibm.autocvt property
		 * determines whether we convert input file I/O based on file tagging. If not explicitly specified,
		 * the property defaults to true, unless file.encoding is set to COMPAT.
		 */
		zOSAutoConvert = getSysPropBeforePropertiesInitialized(sysPropID_zOSAutoConvert);
		/*[ENDIF] (JAVA_SPEC_VERSION >= 21) & (PLATFORM-mz31 | PLATFORM-mz64) */
```
The same condition should be used on lines 733-737:
```java
	/*[IF (JAVA_SPEC_VERSION >= 21) & (PLATFORM-mz31 | PLATFORM-mz64)]*/
	initializedProperties.put("com.ibm.autocvt", zOSAutoConvert); //$NON-NLS-1$
	/*[ENDIF] (JAVA_SPEC_VERSION >= 21) & (PLATFORM-mz31 | PLATFORM-mz64) */
```